ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/422__model_serving_llm_other_runtimes_UI.robot

```diff
@@ -0,0 +1,99 @@
+Resource          ../../../../Resources/CLI/ModelServing/llm.resource
+Suite Setup       Non-Admin Setup Kserve UI Test
+Suite Teardown    Non-Admin Teardown Kserve UI Test
+Test Tags         KServe
+
+
+*** Variables ***
+${LLM_RESOURCES_DIRPATH}=    ods_ci/tests/Resources/Files/llm
+${TEST_NS}=    runtimes-ui
+${EXP_RESPONSES_FILEPATH}=    ${LLM_RESOURCES_DIRPATH}/model_expected_responses.json
+${FLAN_MODEL_S3_DIR}=    flan-t5-small/flan-t5-small-hf
+${FLAN_STORAGE_URI}=    s3://${S3.BUCKET_3.NAME}/${FLAN_MODEL_S3_DIR}/
+${TGIS_RUNTIME_NAME}=    tgis-runtime
+@{SEARCH_METRICS}=    tgi_    istio_
+
+
+*** Test Cases ***
+Verify Non Admin Can Serve And Query A Model Using The UI  # robocop: disable
+    [Documentation]    Basic tests leveraging on a non-admin user for preparing, deploying and querying a LLM model
+    ...                using Single-model platform and TGIS Standalone runtime.
+    [Tags]    Sanity    Tier1    ODS-2611
+    [Setup]    Run    git clone https://github.com/IBM/text-generation-inference/
+    ${test_namespace}=    Set Variable     ${TEST_NS}
+    ${model_name}=    Set Variable    flan-t5-small-hf
+    Deploy Kserve Model Via UI    model_name=${model_name}
+    ...    serving_runtime=TGIS Standalone ServingRuntime for KServe (gRPC)
+    ...    data_connection=kserve-connection    model_framework=pytorch    path=${FLAN_MODEL_S3_DIR}
+    Wait For Pods To Be Ready    label_selector=serving.kserve.io/inferenceservice=${model_name}
+    ...    namespace=${test_namespace}
+    Query Model Multiple Times    model_name=${model_name}    runtime=${TGIS_RUNTIME_NAME}
+    ...    inference_type=all-tokens    n_times=1
+    ...    namespace=${test_namespace}    protocol=grpc
+    Query Model Multiple Times    model_name=${model_name}        runtime=${TGIS_RUNTIME_NAME}
+    ...    inference_type=streaming    n_times=1
+    ...    namespace=${test_namespace}    protocol=grpc    validate_response=${FALSE}
+    Wait Until Keyword Succeeds    30 times    4s
+    ...    Metrics Should Exist In UserWorkloadMonitoring
+    ...    thanos_url=${THANOS_URL}    thanos_token=${THANOS_TOKEN}
+    ...    search_metrics=${SEARCH_METRICS}
+    Wait Until Keyword Succeeds    50 times    5s
+    ...    User Can Fetch Number Of Requests Over Defined Time
+    ...    thanos_url=${THANOS_URL}    thanos_token=${THANOS_TOKEN}
+    ...    model_name=${model_name}    query_kind=single    namespace=${test_namespace}    period=5m    exp_value=1
+    Delete Model Via UI    ${model_name}
+
+Verify Model Can Be Served And Query On A GPU Node Using The UI  # robocop: disable
+    [Documentation]    Basic tests for preparing, deploying and querying a LLM model on GPU node
+    ...                using Single-model platform and TGIS Standalone runtime.
+    [Tags]    Sanity    Tier1    ODS-2612   Resources-GPU
+    [Setup]    Run    git clone https://github.com/IBM/text-generation-inference/
```

Comment:
another clone of `text-generation-inference`...

Reply:
yes. The idea around this suite is to put tests about more basic tests for serving runtimes which will come. Not all of them will need the same repos. So I clone it again here